File: spec/project_euler/problem_1_spec.rb
require 'spec_helper'

describe Problem1 do
  it "multiples of 3 and 5 should give 23 under 10" do
    Problem1.new.execute(10).should eq 23
  end

  it "multiples of 3 and 5 should give 233168 under 1000" do
    Problem1.new.execute(1000).should eq 233168
  end
end

File: scala/src/test/scala/project_euler/problems/Problem1Spec.scala
package project_euler.problems

import org.junit.runner.RunWith
import org.scalatest.FunSpec
import org.scalatest.junit.JUnitRunner
import org.scalatest.Matchers


@RunWith(classOf[JUnitRunner])
class Problem1Spec extends FunSpec with Matchers {
  describe ("#calculate") {
    it ("returns 10 for everything") {
      new Problem1().execute should equal(10)
    }
  }
}
